clients/src/main/java/org/apache/kafka/common/serialization/Serdes.java

```diff
@@ -0,0 +1,155 @@
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package org.apache.kafka.common.serialization;
+
+/**
+ * Factory for creating serializers / deserializers.
+ */
+public class Serdes {
+
+    static public final class LongSerde implements Serde<Long> {
+        @Override
+        public Serializer<Long> serializer() {
+            return new LongSerializer();
+        }
+
+        @Override
+        public Deserializer<Long> deserializer() {
+            return new LongDeserializer();
+        }
+    }
+
+    static public final class IntegerSerde implements Serde<Integer> {
+        @Override
+        public Serializer<Integer> serializer() {
+            return new IntegerSerializer();
+        }
+
+        @Override
+        public Deserializer<Integer> deserializer() {
+            return new IntegerDeserializer();
+        }
+    }
+
+    static public final class DoubleSerde implements Serde<Double> {
+        @Override
+        public Serializer<Double> serializer() {
+            return new DoubleSerializer();
+        }
+
+        @Override
+        public Deserializer<Double> deserializer() {
+            return new DoubleDeserializer();
+        }
+    }
+
+    static public final class StringSerde implements Serde<String> {
+        @Override
+        public Serializer<String> serializer() {
+            return new StringSerializer();
+        }
+
+        @Override
+        public Deserializer<String> deserializer() {
+            return new StringDeserializer();
+        }
+    }
+
+    static public final class ByteArraySerde implements Serde<byte[]> {
+        @Override
+        public Serializer<byte[]> serializer() {
+            return new ByteArraySerializer();
+        }
+
+        @Override
+        public Deserializer<byte[]> deserializer() {
+            return new ByteArrayDeserializer();
+        }
+    }
+
+    @SuppressWarnings("unchecked")
+    static public <T> Serde<T> serdeFrom(Class<T> type) {
+        if (String.class.isAssignableFrom(type)) {
+            return (Serde<T>) String();
+        }
+
+        if (Integer.class.isAssignableFrom(type)) {
+            return (Serde<T>) Integer();
+        }
+
+        if (Long.class.isAssignableFrom(type)) {
+            return (Serde<T>) Long();
+        }
+
+        if (Double.class.isAssignableFrom(type)) {
+            return (Serde<T>) Double();
+        }
+
+        if (byte[].class.isAssignableFrom(type)) {
+            return (Serde<T>) ByteArray();
+        }
+
+        // TODO: we can also serializes objects of type T using generic Java serialization by default
+        throw new IllegalArgumentException("Unknown class for built-in serializer");
+    }
+
+    static public <T> Serde<T> serdeFrom(final Serializer<T> serializer, final Deserializer<T> deserializer) {
```

Comment:
We should add a docstring.  Notably what the contract is if you pass in `null` for any of its arguments.  But see my note below on how I think we should/need to improve the `serdeFrom` functionality by explicitly having separate functions for constructing partial serdes.  (This is a situation where I really miss Scala :-P)